### DISCUSSION TITLE
Problems using @layer in separate css files

### DISCUSSION THREAD

**Discussion by djdanne** [tailwindlabs/tailwindcss#10036] [Help] [answered]
I have a problem whenever I try to use tailwind directives outside the same file where I import the tailwind layers I get the following error: 
```css
`@layer base` is used but no matching `@tailwind base` directive is present.
```
As long as I use @layer inside the same file as I import the tailwind layers everything works fine.

My project is a React project bootstrapped with create-react-app 5 and using the latest tailwindcss (3.1.8), postcss (8.2.4) and postcss-import (15.0.0) and I have followed the guides for setup of this: https://tailwindcss.com/docs/guides/create-react-app and https://tailwindcss.com/docs/using-with-preprocessors.

I have looked around on others having the same problem, but in most cases it looks like some missconfiguration of postcss-import (for example https://github.com/tailwindlabs/tailwindcss/discussions/5934), but in my project I havn't been able to find the potential missconfiguration. 

I hope someone can come with a suggestion on how to make it work, it would be nice to be able to have it in separate css files :)

Some example code from my project:
index.css (main entry point):
```css
@import 'tailwindcss/base';

@import 'tailwindcss/components';
@import './components/input';
/* more imported components */

@import 'tailwindcss/utilities';
```

input.css:
```css
/* This doesn't work, if I move it to index.css it works */
@layer base {
  input,
  textarea {
    @apply placeholder-fb-pink; /* fb-pink is a custom tailwind color */
  }
}
```

postcss.config.js:
```javascript
module.exports = {
  plugins: {
    'postcss-import': {},
    'tailwindcss/nesting': {},
    tailwindcss: { config: './tailwindcss.config.js' },
    autoprefixer: {},
  },
};
```

**Comment by adamwathan** [comment, marked as answer]
Hey! Issue here is create react app, it ignores your PostCSS config file so you aren’t actually using postcss-import even though you’re trying to. We recommend against using CRA for this reason, it’s very non-configurable.

**Reply by djdanne** [reply]
I'm aware of the limitations of CRA (and looking at alternatives), but I got fooled by step 2 in https://tailwindcss.com/docs/guides/create-react-app, since it actually creates a postcss.config.js. I therefore assumed that CRA supported external postcss configs, but now I realize it does nothing (tried to remove it from my project). I guess it just works because CRA already uses postcss and autoprefixer behind the scenes?

But in order to avoid confusion around this, maybe the text about "more control" in https://tailwindcss.com/docs/guides/create-react-app could be clarified, at least with that the postcss config is not used?

Thanks for the quick response!
